### PR TITLE
DNS Delegation Support for Bloxone Ansible v2

### PR DESCRIPTION
**Pull request by AnilGadiyarHJ**

### CODE REVIEW

File: plugins/module_utils/modules.py [hunk 1]
```diff
@@ -115,6 +115,14 @@ def _is_changed(existing, payload):
         if v is not None:
             if k not in existing:
                 changed = True
+            elif isinstance(v, list):
+                # If the order of the list is different, it is considered as changed
+                if len(v) != len(existing[k]):
+                    changed = True
+                else:
```
Comment (JchhatbarInfoblox) [inline plugins/module_utils/modules.py:122]: remove else we should check nested list as well.

Comment (unasra) [reply plugins/module_utils/modules.py:122]: The Else condition provided in the loop below recursively checks if the existing attributes is equal to the payload (or the subset of the payload) , hence it is vital to the elif appended here.
Also , if the length of the existing attribute doesnt match the payload , there is no point of checking the list and we can directly send changed = True.

File: plugins/modules/dns_delegation.py [hunk 2]
```diff
@@ -0,0 +1,331 @@
...
+
+EXAMPLES = r"""
+- name: Create an Auth Zone
+  infoblox.bloxone.dns_auth_zone:
+    name: example_zone
+    state: present
+
+- name: Create a Delegation
+  infoblox.bloxone.dns_delegation:
+    fqdn: test_delegation.example_zone.
```
Comment (unasra) [inline plugins/modules/dns_delegation.py]: Examples should not have the keyword "test"

File: plugins/modules/dns_delegation.py [hunk 3]
```diff
@@ -0,0 +1,331 @@
...
+    delegation_servers:
+      - fqdn: ns1.example.com.
+        address: 12.0.0.0
+    state: present
+    tags:
+      location: my-location
+
+- name: Delete the DNS Delegation
+  infoblox.bloxone.dns_delegation:
+    fqdn: test_delegation.example_zone.
```
Comment (unasra) [inline plugins/modules/dns_delegation.py]: "Test" keyword to be removed.

File: plugins/modules/dns_delegation_info.py [hunk 4]
```diff
@@ -0,0 +1,254 @@
...
+        id: '{{ delegation_fqdn_id }}'
+        
+    - name: Get Delegation information by filters (e.g. fqdn)
+      infoblox.bloxone.dns_delegation_info:
+        filters:
+          fqdn: "delegation.example_zone."
+
+    - name: Get Delegation information by raw filter query
+      infoblox.bloxone.dns_delegation_info:
+        filter_query: name=='delegation.example_zone'
```
Comment (unasra) [inline plugins/modules/dns_delegation_info.py]: Does delegation have a name parameter ?

File: plugins/modules/dns_delegation_info.py [hunk 5]
```diff
@@ -0,0 +1,254 @@
...
+        description:
+            - Filter dict to filter objects
+        type: dict
+        required: false
+    filter_query:
+        description:
+            - Filter query to filter objects
+        type: str
+        required: false
+    inherit:
```
Comment (unasra) [inline plugins/modules/dns_delegation_info.py]: Does delegation support inheritance?
If yes , then support needs to be added in the dns_delegation module !

File: plugins/modules/dns_delegation_info.py [hunk 6]
```diff
@@ -0,0 +1,254 @@
...
+        self.exit_json(**result)
+
+
+def main():
+    # define available arguments/parameters a user can pass to the module
+    module_args = dict(
+        id=dict(type="str", required=False),
+        filters=dict(type="dict", required=False),
+        filter_query=dict(type="str", required=False),
+        inherit=dict(type="str", required=False, choices=["full", "partial", "none"], default="full"),
```
Comment (unasra) [inline plugins/modules/dns_delegation_info.py]: Is inheritance supported ?

File: tests/integration/targets/dns_delegation/meta/main.yml [hunk 7]
```diff
@@ -0,0 +1,2 @@
+---
+dependencies: [setup_view]
```
Comment (unasra) [inline tests/integration/targets/dns_delegation/meta/main.yml]: Add a line at the end of the file.

File: tests/integration/targets/dns_delegation/tasks/main.yml [hunk 8]
```diff
@@ -0,0 +1,277 @@
+- module_defaults:
```
Comment (unasra) [inline tests/integration/targets/dns_delegation/tasks/main.yml:2]: Its not a compulsion , but since Ansible playbooks are written in YAML, it might be a good idea to start the document with "---"

File: tests/integration/targets/dns_delegation/tasks/main.yml [hunk 9]
```diff
@@ -0,0 +1,277 @@
...
+      api_key: "{{ api_key }}"
+
+  block:
+    # Create a random Delegation name to avoid conflicts
+    - ansible.builtin.set_fact:
+        auth_zone_fqdn: "test-auth-zone-{{ 999999 | random | string }}.com."
+    - ansible.builtin.set_fact:
+        delegation_fqdn: "test-delegation--{{ 999999 | random | string }}.{{ auth_zone_fqdn }}"
+
+    - name: Create an Auth Zone in the specified view
```
Comment (unasra) [inline tests/integration/targets/dns_delegation/tasks/main.yml]: Add a setup for auth zone ( similar to  the way  setup_view is done )

File: tests/integration/targets/dns_delegation/tasks/main.yml [hunk 10]
```diff
@@ -0,0 +1,277 @@
...
+          fqdn: "{{ delegation_fqdn }}"
+      register: delegation_info
+    - assert:
+        that:
+          - delegation is changed
+          - delegation_info is not failed
+          - delegation_info.objects | length == 0
+
+
+    # Create a Delegation
```
Comment (unasra) [inline tests/integration/targets/dns_delegation/tasks/main.yml]: The name conveys the exact same information as the comment , so the comments provided might be redundant.

File: tests/integration/targets/dns_delegation_info/meta/main.yml [hunk 11]
```diff
@@ -0,0 +1,2 @@
+---
+dependencies: [setup_view]
```
Comment (unasra) [inline tests/integration/targets/dns_delegation_info/meta/main.yml]: Add a new line at the end of the file .

File: tests/integration/targets/dns_delegation_info/tasks/main.yml [hunk 12]
```diff
@@ -0,0 +1,90 @@
...
+      api_key: "{{ api_key }}"
+  block:
+    # Create a random Delegation name to avoid conflicts
+    - ansible.builtin.set_fact:
+        auth_zone_fqdn: "test-auth-zone-{{ 999999 | random | string }}.com."
+        tag_value: "site-{{ 999999 | random | string }}"
+    - ansible.builtin.set_fact:
+        delegation_fqdn: "test-delegation--{{ 999999 | random | string }}.{{ auth_zone_fqdn }}"
+
+    - name: Create an Auth Zone in the specified view
```
Comment (unasra) [inline tests/integration/targets/dns_delegation_info/tasks/main.yml]: Use setup_auth_zone instead.

File: tests/integration/targets/dns_delegation_info/tasks/main.yml [hunk 13]
```diff
@@ -0,0 +1,90 @@
...
+      infoblox.bloxone.dns_delegation_info:
+        filters:
+          fqdn: "{{ delegation_fqdn }}"
+      register: delegation_info
+    - assert:
+        that:
+          - delegation_info.objects | length == 1
+          - delegation_info.objects[0].fqdn == delegation.object.fqdn
+
+    - name: Get Delegation information by filters (FQDN)
```
Comment (unasra) [inline tests/integration/targets/dns_delegation_info/tasks/main.yml:24]: Same test as above

File: tests/integration/targets/dns_delegation/tasks/main.yml [hunk 14]
```diff
@@ -0,0 +1,277 @@
...
+    - assert:
+        that:
+          - delegation_info is not failed
+          - delegation_info.objects | length == 1
+          - delegation_info.objects[0].id == delegation.id
+          - delegation_info.objects[0].fqdn == delegation_fqdn
+          - delegation_info.objects[0].comment == "Delegation zone created with Ansible for tags test"
+          - delegation_info.objects[0].tags.location == "site-1"
+
+    - name: Create DNS Delegation in view
```
Comment (unasra) [inline tests/integration/targets/dns_delegation/tasks/main.yml]: Since the view isnt being changed here , it might just be a good idea to add the check in any other UT

File: plugins/modules/dns_delegation.py (hunk 2 above)
Comment (AnilGadiyarHJ) [reply plugins/modules/dns_delegation.py]: done

File: plugins/modules/dns_delegation.py (hunk 3 above)
Comment (AnilGadiyarHJ) [reply plugins/modules/dns_delegation.py]: done

File: plugins/modules/dns_delegation_info.py (hunk 4 above)
Comment (AnilGadiyarHJ) [reply plugins/modules/dns_delegation_info.py]: changed to FQDN

File: plugins/modules/dns_delegation_info.py (hunk 5 above)
Comment (AnilGadiyarHJ) [reply plugins/modules/dns_delegation_info.py]: it doesn't support inheritance

File: plugins/modules/dns_delegation_info.py (hunk 6 above)
Comment (AnilGadiyarHJ) [reply plugins/modules/dns_delegation_info.py]: No, should i remove the line?

File: tests/integration/targets/dns_delegation/meta/main.yml (hunk 7 above)
Comment (AnilGadiyarHJ) [reply tests/integration/targets/dns_delegation/meta/main.yml]: done

File: tests/integration/targets/dns_delegation/tasks/main.yml (hunk 8 above)
Comment (AnilGadiyarHJ) [reply tests/integration/targets/dns_delegation/tasks/main.yml:2]: done

File: tests/integration/targets/dns_delegation/tasks/main.yml (hunk 9 above)
Comment (AnilGadiyarHJ) [reply tests/integration/targets/dns_delegation/tasks/main.yml]: done

File: tests/integration/targets/dns_delegation/tasks/main.yml (hunk 10 above)
Comment (AnilGadiyarHJ) [reply tests/integration/targets/dns_delegation/tasks/main.yml]: done

File: tests/integration/targets/dns_delegation_info/meta/main.yml (hunk 11 above)
Comment (AnilGadiyarHJ) [reply tests/integration/targets/dns_delegation_info/meta/main.yml]: done

File: tests/integration/targets/dns_delegation_info/tasks/main.yml (hunk 12 above)
Comment (AnilGadiyarHJ) [reply tests/integration/targets/dns_delegation_info/tasks/main.yml]: done

File: plugins/modules/dns_delegation_info.py (hunk 6 above)
Comment (unasra) [reply plugins/modules/dns_delegation_info.py]: Yes , remove the line.

File: tests/integration/targets/dns_delegation/meta/main.yml [hunk 15]
```diff
@@ -0,0 +1,2 @@
+---
+dependencies: [setup_auth_zone]
```
Comment (unasra) [inline tests/integration/targets/dns_delegation/meta/main.yml]: Missing line at the end of the file.

File: plugins/modules/dns_delegation.py [hunk 16]
```diff
@@ -0,0 +1,331 @@
...
+        type: str
+
+extends_documentation_fragment:
+    - infoblox.bloxone.common
+"""  # noqa: E501
+
+EXAMPLES = r"""
+- name: Create an Auth Zone
+  infoblox.bloxone.dns_auth_zone:
+    name: example_zone
```
Comment (unasra) [inline plugins/modules/dns_delegation.py]: Does auth zone have a name parameter ?

File: plugins/modules/dns_delegation.py [hunk 17]
```diff
@@ -0,0 +1,331 @@
...
+    view:
+        description:
+            - "The resource identifier."
+        type: str
+
+extends_documentation_fragment:
+    - infoblox.bloxone.common
+"""  # noqa: E501
+
+EXAMPLES = r"""
```
Comment (unasra) [inline plugins/modules/dns_delegation.py:79]: Please Test the examples provided here .

File: tests/integration/targets/dns_delegation/tasks/main.yml [hunk 18]
```diff
@@ -0,0 +1,267 @@
...
+        filters:
+          fqdn: "{{ delegation_fqdn }}"
+      register: delegation_info
+    - assert:
+        that:
+          - delegation is changed
+          - delegation_info.objects | length == 1
+          - delegation_info.objects[0].fqdn == delegation.object.fqdn
+
+    # Create a Delegation (idempotent)
```
Comment (unasra) [inline tests/integration/targets/dns_delegation/tasks/main.yml]: Unnecessary comment ( same for the 3 TC's below .

File: tests/integration/targets/dns_delegation_info/tasks/main.yml (hunk 13 above)
Comment (AnilGadiyarHJ) [reply tests/integration/targets/dns_delegation_info/tasks/main.yml:24]: done

File: plugins/modules/dns_delegation_info.py (hunk 6 above)
Comment (AnilGadiyarHJ) [reply plugins/modules/dns_delegation_info.py]: done

File: tests/integration/targets/dns_delegation/tasks/main.yml (hunk 14 above)
Comment (AnilGadiyarHJ) [reply tests/integration/targets/dns_delegation/tasks/main.yml]: done, removed UT and added the check in the beginning

File: tests/integration/targets/dns_delegation_info/meta/main.yml [hunk 19]
```diff
@@ -0,0 +1,2 @@
+---
+dependencies: [setup_auth_zone]
```
Comment (unasra) [inline tests/integration/targets/dns_delegation_info/meta/main.yml]: Add a line at the end of file

File: tests/integration/targets/dns_delegation/meta/main.yml (hunk 15 above)
Comment (unasra) [inline tests/integration/targets/dns_delegation/meta/main.yml]: Where is setup auth zone defined ?
Please run integration and sanity tests before every push.

Comment (AnilGadiyarHJ) [reply tests/integration/targets/dns_delegation/meta/main.yml]: missed committing the setup auth zone, commit it now
integration and sanity tests are passing 
Screenshot 2024-11-20 at 3 59 34 PM

File: tests/integration/targets/dns_delegation_info/meta/main.yml (hunk 19 above)
Comment (AnilGadiyarHJ) [reply tests/integration/targets/dns_delegation_info/meta/main.yml]: done

File: tests/integration/targets/dns_delegation/tasks/main.yml (hunk 18 above)
Comment (AnilGadiyarHJ) [reply tests/integration/targets/dns_delegation/tasks/main.yml]: removed

File: plugins/modules/dns_delegation.py (hunk 16 above)
Comment (AnilGadiyarHJ) [reply plugins/modules/dns_delegation.py]: no, it doesn't, I have corrected the same in dns_auth_zone.py as well

File: tests/integration/targets/dns_delegation/meta/main.yml (hunk 15 above)
Comment (AnilGadiyarHJ) [reply tests/integration/targets/dns_delegation/meta/main.yml]: done

File: plugins/modules/dns_delegation.py (hunk 17 above)
Comment (AnilGadiyarHJ) [reply plugins/modules/dns_delegation.py:79]: Done

File: plugins/modules/dns_auth_zone.py [hunk 20]
```diff
@@ -630,12 +630,12 @@
 EXAMPLES = r"""
   - name: Create an Auth Zone
     infoblox.bloxone.dns_auth_zone:
-      name: "example_zone"
+      fqdn: "example_zone"
```
Comment (unasra) [inline plugins/modules/dns_auth_zone.py]: Lets keep this PR for Delegation , I will raise a separate PR for this

File: tests/integration/targets/dns_delegation/tasks/main.yml [hunk 21]
```diff
@@ -0,0 +1,243 @@
...
+
+    - name: "Clean up Authorization Zone"
+      infoblox.bloxone.dns_auth_zone:
+        fqdn: "{{ _fqdn_auth_zone }}"
+        state: absent
+      ignore_errors: true
+
+    - name: "Clean up  DNS View"
+      infoblox.bloxone.dns_view:
+        name: "{{ _view.id }}"
```
Comment (unasra) [inline tests/integration/targets/dns_delegation/tasks/main.yml]: Why is view name being assigned to Id here ?

File: tests/integration/targets/setup_auth_zone/meta/main.yml [hunk 22]
```diff
@@ -0,0 +1,2 @@
+---
+dependencies: [setup_view]
```
Comment (unasra) [inline tests/integration/targets/setup_auth_zone/meta/main.yml]: Missing line at the end of the file .

File: tests/integration/targets/setup_auth_zone/tasks/main.yml [hunk 23]
```diff
@@ -0,0 +1,17 @@
...
+    - ansible.builtin.set_fact:
+        _fqdn_auth_zone: "test-auth-zone-{{ 999999 | random | string }}.com."
+
+    - name: Create an Auth Zone in a view
+      infoblox.bloxone.dns_auth_zone:
+        fqdn: "{{ _fqdn_auth_zone }}"
+        view:  "{{ _view.id }}"
+        primary_type: cloud
+        state: present
+      register: _auth_zone
```
Comment (unasra) [inline tests/integration/targets/setup_auth_zone/tasks/main.yml]: Missing line at the end of the file.

File: tests/integration/targets/dns_delegation_info/tasks/main.yml [hunk 24]
```diff
@@ -0,0 +1,70 @@
...
+
+    - name: "Clean up Authorization Zone"
+      infoblox.bloxone.dns_auth_zone:
+        fqdn: "{{ _fqdn_auth_zone }}"
+        state: absent
+      ignore_errors: true
+
+    - name: "Clean up  DNS View"
+      infoblox.bloxone.dns_view:
+        name: "{{ _view.id }}"
```
Comment (unasra) [inline tests/integration/targets/dns_delegation_info/tasks/main.yml]: View name being assigned to id

File: tests/integration/targets/dns_delegation/meta/main.yml (hunk 15 above)
Comment (AnilGadiyarHJ) [reply tests/integration/targets/dns_delegation/meta/main.yml]: done

File: plugins/modules/dns_auth_zone.py (hunk 20 above)
Comment (AnilGadiyarHJ) [reply plugins/modules/dns_auth_zone.py]: i have reverted the change

File: tests/integration/targets/dns_delegation/tasks/main.yml (hunk 21 above)
Comment (AnilGadiyarHJ) [reply tests/integration/targets/dns_delegation/tasks/main.yml]: corrected this and tested

File: tests/integration/targets/setup_auth_zone/meta/main.yml (hunk 22 above)
Comment (AnilGadiyarHJ) [reply tests/integration/targets/setup_auth_zone/meta/main.yml]: done

File: tests/integration/targets/setup_auth_zone/tasks/main.yml (hunk 23 above)
Comment (AnilGadiyarHJ) [reply tests/integration/targets/setup_auth_zone/tasks/main.yml]: done

File: tests/integration/targets/dns_delegation_info/tasks/main.yml (hunk 24 above)
Comment (AnilGadiyarHJ) [reply tests/integration/targets/dns_delegation_info/tasks/main.yml]: corrected this and tested

**Comment by unasra** [comment]
LGTM 

@mathewab please have a look.

File: plugins/modules/dns_delegation.py [hunk 25]
```diff
@@ -0,0 +1,332 @@
...
+- name: Create an Auth Zone
+  infoblox.bloxone.dns_auth_zone:
```
Comment (mathewab) [inline plugins/modules/dns_delegation.py:81]: do we need an auth_zone example in the delegation module?

File: plugins/modules/dns_delegation.py [hunk 26]
```diff
@@ -0,0 +1,332 @@
...
+        if self.params["id"] is not None:
+            try:
+                resp = DelegationApi(self.client).read(self.params["id"], inherit="full")
+                return resp.result
+            except NotFoundException as e:
+                if self.params["state"] == "absent":
+                    return None
+                raise e
+        else:
+            filter = f"fqdn=='{self.params['fqdn']}'"
```
Comment (mathewab) [inline plugins/modules/dns_delegation.py]: shouldn't this also include `view`?

File: plugins/modules/dns_delegation_info.py [hunk 27]
```diff
@@ -0,0 +1,243 @@
...
+
+EXAMPLES = r"""
+    - name: Get Information about the Delegation
+      infoblox.bloxone.dns_delegation_info:
+        id: '{{ delegation_fqdn_id }}'
+        
+    - name: Get Delegation information by filters (e.g. fqdn)
+      infoblox.bloxone.dns_delegation_info:
+        filters:
+          fqdn: "delegation.example_zone."
```
Comment (mathewab) [inline plugins/modules/dns_delegation_info.py:57]: can you also add an example for a delegation not in a default view

File: tests/integration/targets/dns_delegation_info/tasks/main.yml [hunk 28]
```diff
@@ -0,0 +1,70 @@
...
+      infoblox.bloxone.dns_auth_zone:
+        fqdn: "{{ _fqdn_auth_zone }}"
+        state: absent
+      ignore_errors: true
+
+    - name: "Clean up  DNS View"
+      infoblox.bloxone.dns_view:
+        name: "{{ name }}"
+        state: absent
+      ignore_errors: true
```
Comment (mathewab) [inline tests/integration/targets/dns_delegation_info/tasks/main.yml:72]: Don't have to do it in this PR, but I believe it should be possible to have a cleanup.yml file in `setup_view` and call that instead. 

```
- ansible.builtin.include_role:
    name: setup_view
    tasks_from: cleanup.yml
```

File: plugins/modules/dns_delegation.py (hunk 25 above)
Comment (AnilGadiyarHJ) [reply plugins/modules/dns_delegation.py:81]: Delegation creation needs Auth zone, I feel it should be included in the example

File: tests/integration/targets/dns_delegation_info/tasks/main.yml (hunk 28 above)
Comment (AnilGadiyarHJ) [reply tests/integration/targets/dns_delegation_info/tasks/main.yml:72]: Noted

File: plugins/modules/dns_delegation.py (hunk 26 above)
Comment (AnilGadiyarHJ) [reply plugins/modules/dns_delegation.py]: Since view is not a required parameter for delegation, Not every user might create a custom view, some might use the Default view for the same and I think we can't add a filter if the view is not provided beforehand, hence it wasn't added to the  filter

Comment (unasra) [reply plugins/modules/dns_delegation.py]: If we were to implement this , 
This could have two cases : Where user provides the view and when he doesnt 

Where user provides the view , this is sorted . 

But when view is no provided , we should assume that delegation is intended to be created in a zone in the default view . 
so we would have to make a view list call , provide default as a name filter and retrieve the view id.  
Optionally we can also check if the zone is present in the default view , coz if we dont check and zone isnt present , the backend would return "record not found".
Once this is verified , then the filter query would be executed with the view set

File: plugins/modules/dns_delegation.py (hunk 25 above)
Comment (mathewab) [reply plugins/modules/dns_delegation.py:81]: I don't think you need a zone to create a delegation in API. you should be able to create a delegation by simply providing the fqdn.
(oddly, in UI there is a required parent field)

File: plugins/modules/dns_delegation.py (hunk 26 above)
Comment (mathewab) [reply plugins/modules/dns_delegation.py]: Yes, I agree for idempotency, it should always include the view in the `find` filter. There are two approaches that you can take to it. 
- As you said, if `view` is not given by user, we can retrieve default view and use that ID. 
- Make view as a required parameter for all types of zones. Providing view_id of default view will have to be the user's responsibility in this case. I think this is much better, because the user can be explicit about which zone is being updated especially if there is one with the same fqdn in many views.

File: plugins/modules/dns_delegation.py (hunk 25 above)
Comment (AnilGadiyarHJ) [reply plugins/modules/dns_delegation.py:81]: delegation gets created inside the primary zone, I don't see an option to make it without one.  please suggest should I keep it or remove ?

Comment (mathewab) [reply plugins/modules/dns_delegation.py:81]: Sorry, I may have written incorrectly before. Let me rephrase it. You don't need to provide a zone_id to create a delegation in the API. It looks closest matching parent from the fqdn. 

I checked the TF documentation. It looks like we have auth_zone there in the example as well. 
We can keep it. But i would add a comment `Create an Auth Zone (Delegation requires a parent zone)

File: plugins/modules/dns_delegation.py (hunk 26 above)
Comment (unasra) [reply plugins/modules/dns_delegation.py:230]: Second method would definitely be better , but this would fail if the user fails to create a matching parent zone or has a typo in the delegation fqdn , coz if the matching Auth Zone isnt present in View. , then the backend would prompt the error "The delegation must have a parent zone" .

Comment (mathewab) [reply plugins/modules/dns_delegation.py]: Why would that be an issue? I think the error correctly identifies that the given fqdn has an issue finding the parent zone. 

Infact, I feel the first approach might not work for idempotency. For example,  I have this auth_zone task that creates in the default view. Next day I create another view and create the same zone manually. Coming back to this task and replaying would now give me an error. 

```
    - name: Create an Auth Zone (check mode)
      infoblox.bloxone.dns_auth_zone:
        fqdn: "{{ fqdn }}"
        primary_type: external
        state: present
```

File: plugins/modules/dns_delegation_info.py (hunk 27 above)
Comment (AnilGadiyarHJ) [reply plugins/modules/dns_delegation_info.py:57]: done

File: plugins/modules/dns_delegation.py (hunk 25 above)
Comment (AnilGadiyarHJ) [reply plugins/modules/dns_delegation.py:81]: done

File: plugins/modules/dns_delegation.py (hunk 26 above)
Comment (AnilGadiyarHJ) [reply plugins/modules/dns_delegation.py]: We tried including view as a mandatory field and added it to the filters along with FQDN. below backend issue was noticed
https://infoblox.atlassian.net/browse/NORTHSTAR-12614

Comment (AnilGadiyarHJ) [reply plugins/modules/dns_delegation.py]: Thanks for the quick backend fix, I've tested it and filter with view, and fqdn works

**Review by mathewab — APPROVED** [review]
